apps/src/code-studio/pd/workshop_dashboard/reports/foorm/results.jsx

```diff
@@ -0,0 +1,154 @@
+import PropTypes from 'prop-types';
+import React from 'react';
+import {Tab, Tabs} from 'react-bootstrap';
+import ChoiceResponses from '../../components/survey_results/choice_responses';
+import TextResponses from '../../components/survey_results/text_responses';
+import _ from 'lodash';
+
+export default class Results extends React.Component {
+  static propTypes = {
+    questions: PropTypes.object.isRequired,
+    thisWorkshop: PropTypes.object.isRequired,
+    sessions: PropTypes.arrayOf(PropTypes.string).isRequired,
+    courseName: PropTypes.string
+    // workshopRollups: PropTypes.object,
+    // facilitatorRollups: PropTypes.object
+  };
+
+  /**
+   * Render results for either the facilitator specific or general questions
+   * section: name of section
+   * questions: all questions, keyed by survey name
+   * answers: hash of {survey_name : {answers hash}}
+   */
+  renderResultsForSessionQuestionSection(section, questions, answers) {
+    let uniqueKey = -1;
+    return _.compact(
+      Object.keys(answers).map(surveyId => {
+        let surveyQuestions = questions[surveyId];
+        if (!surveyQuestions) {
+          return null;
+        }
+
+        return _.compact(
+          Object.keys(surveyQuestions).map(questionId => {
+            let question = surveyQuestions[questionId];
+            let answer = answers[surveyId][questionId];
+            if (!answer) {
+              return null;
+            }
+
+            if (
+              ['scale', 'singleSelect', 'multiSelect'].includes(
+                question['type']
+              )
+            ) {
+              // numRespondents will get either a value (for multiSelect) or undefined.
+              const numRespondents = answer.num_respondents;
+
+              // Make a copy of the answers without the num_respondents and other_answers fields.
+              const filteredAnswers = _.omit(answer, [
+                'num_respondents',
+                'other_answers'
+              ]);
```

Comment:
Should this kind of filtering be done on the server, which presumably would have a bit more of the domain-specific knowledge?

Reply:
the reason this is done is the answer has the format:
```
{
   "num_respondents": 5,
   "other_answers":["abc", "def"],
   "answer_1":"yes",
   "answer_2":"no"
}
```
Where num_respondents and other_answers are optional. Then the rest of the answers are for individual items. I toyed with whether to nest the individual answers but it seemed like overkill (also, the num_respondents with the individual answers was borrowed from our previous pipeline). If you think that would be clearer I can update it.
So the UI needs all the data, but only wants to pass the specific answers as a prop.

Reply:
This seems fine.  I'm more concerned with the goal of having a "naive" UI overall, but I think that's under control :)